Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseDataSample.cs

```diff
@@ -69,7 +69,11 @@ public QuickPulseDataSample(QuickPulseDataAccumulator accumulator, IDictionary<s
                 p => (QuickPulseDefaults.CounterOriginalStringMapping.ContainsKey(p.Value.Item1.OriginalString) ? QuickPulseDefaults.CounterOriginalStringMapping[p.Value.Item1.OriginalString] : p.Value.Item1.OriginalString), 
                 p => p.Value.Item2);
 
+            this.TopCpuData = topCpuData.ToArray();
```

Comment:
In this method, it is important to completely decouple the input and the output. The input comes from the accumulator (that's cleared out and reused every two seconds), while the output is stored for later usage (we'll be sending it out to the server). So any coupling between the two is catastrophic, we need to create a new object.